include/eve/module/core/regular/impl/convert.hpp

```diff
@@ -48,9 +55,8 @@ struct convert_lambda
 };
 
 template<product_type T, product_type U>
-requires(kumi::result::flatten_all_t<T>::size()
-         == kumi::result::flatten_all_t<U>::size()) EVE_FORCEINLINE
-    auto convert_(EVE_SUPPORTS(cpu_), T const& v0, eve::as<U>)
+requires(kumi::result::flatten_all_t<T>::size() == kumi::result::flatten_all_t<U>::size())
+EVE_FORCEINLINE auto convert_(EVE_SUPPORTS(cpu_), T const& v0, eve::as<U>)
```

Comment:
Can we do `tuple<logical<int>, int>`? Because if we can, this is not a sufficient restriction.

I don't think you should bother very much - we can fix the sfinae friendlinness if we ever need to.

Reply:
yes, we need something like all of members of in convert to elements of U. 
We can fix that later, main focus here is stopping silly wide->logical and moving avx512 like treatment to x86 files